server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java

```diff
@@ -671,6 +667,14 @@ public boolean deleteBackup(final Long backupId) {
         if (offering == null) {
             throw new CloudRuntimeException("VM backup offering ID " + vm.getBackupOfferingId() + " does not exist");
         }
+        List<Backup> backupsForVm = backupDao.listByVmId(vm.getDataCenterId(), vmId);
+        if (CollectionUtils.isNotEmpty(backupsForVm)) {
```

Comment:
@Pearl1594 is this workaround just for the dummy provider?

Reply:
@rhtyd yes -  this would be for dummy provider ideally

Reply:
I checked this may cause regression for Veeam @Pearl1594 this code is not necessary I think, even for dummy provider. On deletion of a backup, it shouldn't remove from offering - also as long a backup exists we'll never this logic right?

Reply:
This basically removes the VM (deleted) from backup offering - if the last backup of the deleted VM is being removed.